src/CommunityPatch/Patches/Perks/Cunning/Roguery/NegotiatorPatch.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using HarmonyLib;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents.Map;
+using TaleWorlds.Core;
+using static CommunityPatch.HarmonyHelpers;
+
+namespace CommunityPatch.Patches.Perks.Cunning.Roguery {
+
+  public sealed class NegotiatorPatch : PatchBase<NegotiatorPatch> {
+    
+    public override bool Applied { get; protected set; }
+
+    private static readonly MethodInfo TargetMethodInfo = typeof(DefaultBribeCalculationModel).GetMethod(nameof(DefaultBribeCalculationModel.GetBribeToEnterLordsHall), BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly);
+    private static readonly MethodInfo PatchMethodInfo = typeof(NegotiatorPatch).GetMethod(nameof(Postfix), BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly);
+
+    public override IEnumerable<MethodBase> GetMethodsChecked() {
+      yield return TargetMethodInfo;
+    }
+
+    private PerkObject _perk;
+
+    private static readonly byte[][] Hashes = {
+      new byte[] {
+        // e1.1.0.225190
+        0x7C, 0x2F, 0x71, 0x0A, 0x54, 0x2E, 0x4F, 0x9D,
+        0x5E, 0xF6, 0x5B, 0x99, 0xC8, 0x04, 0x2C, 0xE5,
+        0x68, 0x6F, 0x32, 0xAB, 0x37, 0x07, 0x40, 0x56,
+        0x28, 0x32, 0xAE, 0xC6, 0xC6, 0x90, 0x48, 0x8C
+      }
+    };
+
+    public override void Reset()
+      => _perk = PerkObject.FindFirst(x => x.Name.GetID() == "lH8ZMjEo");
+
+    public override bool? IsApplicable(Game game) {
+      if (_perk == null) return false;
+
+      var patchInfo = Harmony.GetPatchInfo(TargetMethodInfo);
+      if (AlreadyPatchedByOthers(patchInfo)) return false;
+
+      var hash = TargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes);
+    }
+
+    public override void Apply(Game game) {
+      if (Applied) return;
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo, postfix: new HarmonyMethod(PatchMethodInfo));
+      Applied = true;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void Postfix(ref int __result) {
+      var scout = Hero.MainHero?.PartyBelongedTo?.EffectiveScout;
+      if (scout == null) return;
+
+      var perk = ActivePatch._perk;
+      if (!scout.GetPerkValue(perk)) return;
+
+      __result = (int) (__result * -perk.PrimaryBonus);
```

Comment:
```suggestion
      __result = (int) (__result * Math.abs(perk.PrimaryBonus));
```

Reply:
What is the value from the primary bonus?

Reply:
> What is the value from the primary bonus?

`-0.5f`

src:

TaleWorlds.CampaignSystem.DefaultPerks.InitializePerks(), e1.1.2

```
// Negotiator perk
this.RogueryNegotiator.Initialize("{=lH8ZMjEo}Negotiator",
                                  "{=LLj618tA}Bribes are 50% cheaper.",
                                  DefaultSkills.Roguery,
                                  this.GetTierCost(6),
                                  this.RogueryBriber,
                                  SkillEffect.PerkRole.Scout,
                                  -0.5f,  // PRIMARY_BONUS
                                  SkillEffect.PerkRole.None,
                                  0.0f,
                                  SkillEffect.EffectIncrementType.AddFactor);
```

Reply:
I doubt they'd ever want bribes to _cost more_ but the behavior of just using negation is _ok_.

If you don't want to use Math.Abs / MathF.Abs here, you should verify that perk.PrimaryBonus <= 0 in IsApplicable.

Reply:
Well, sometimes the perks seem to be additive factors and sometimes multiplicative. Sometimes even in contradiction to the `AddFactor` argument. For example, Bribe master and negotiation both describe "lowering bribes by 50%" but Bribe Master's PrimaryBonus is 0.5 and Negotiators is -0.5. They're also both `AddFactor` type. Or maybe it's a vanilla bug.

Anyway he just multiplies it in the patch and the result shouldn't be negative so I thought Math.Abs would be best.

Reply:
I really liked the idea of Math.Abs. This will make the solution shorter and understandable